api/src/main/java/org/openmrs/module/patientqueueing/api/PatientQueueingService.java

```diff
@@ -9,12 +9,68 @@
  */
 package org.openmrs.module.patientqueueing.api;
 
+import org.openmrs.Location;
+import org.openmrs.Patient;
+import org.openmrs.Provider;
+import org.openmrs.api.APIException;
 import org.openmrs.api.OpenmrsService;
+import org.openmrs.module.patientqueueing.model.PatientQueue;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Date;
+import java.util.List;
 
 /**
  * The main service of this module, which is exposed for other modules. See
  * moduleApplicationContext.xml on how it is wired up.
  */
 public interface PatientQueueingService extends OpenmrsService {
 	
+	/**
+	 * Get a single patient queue record by parameter The queueId can not be null
+	 * 
+	 * @param queueId Id of the PatientQueue to be retrieved
+	 * @return PatientQueue
+	 * @throws APIException
+	 */
+	@Transactional(readOnly = true)
+	public abstract PatientQueue getPatientQueueById(Integer queueId) throws APIException;
+	
+	/**
+	 * Update or Save patientQueue Requires a patientQueue
+	 * 
+	 * @param patientQueue The PatientQueue to be saved
+	 * @return PatientQueue
+	 * @throws APIException
+	 */
+	@Transactional
+	public abstract PatientQueue savePatientQue(PatientQueue patientQueue) throws Exception;
+	
+	/**
+	 * Get List of PatientQueues
+	 * 
+	 * @param provider The provider where the patient was being sent. It Can be null
+	 * @param fromDate lowest date a query will be built upon. It can be null
+	 * @param toDate highest date a query will be built upon. It Can be null
+	 * @param locationTo Location Where patient was sent to
+	 * @param locationFrom Location Where patient was sent from
+	 * @param patient The patient who is in the queued
+	 * @param status Status such as COMPLETED,PENDING
+	 * @return List<PatientQueue>
+	 * @throws APIException
+	 */
+	public List<PatientQueue> getPatientQueueList(Provider provider, Date fromDate, Date toDate, Location locationTo,
+	        Location locationFrom, Patient patient, String status) throws APIException;
+	
+	/**
+	 * Mark patient queue completed. Both Status and the patientQueue to complete are required
+	 * 
+	 * @param patientQueue
+	 * @param status Status such as COMPLETED which is being set towards completion of the
```

Comment:
Do we need to pass the status as a parameters? The method name seems to say it all as status COMPLETED